incubator/kube-downscaler/templates/clusterrole.yaml

```diff
@@ -0,0 +1,32 @@
+{{- if .Values.rbac.create -}}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ include "kube-downscaler.fullname" . }}
+  labels:
+    app: {{ template "kube-downscaler.fullname" . }}
+    helm.sh/chart: {{ include "kube-downscaler.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - pods
+    verbs:
+      - get
+      - watch
+      - list
+  - apiGroups:
+      - extensions
+      - apps
+    resources:
+      - deployments
```

Comment:
We also need the resource namespaces. `kube-downscaler` can read the annotation of a namespace to know if it should downscale all resources in it.

Otherwise you get:

```

2019-05-24 14:12:32,262 DEBUG: https://172.20.0.1:443 "GET /api/v1/namespaces/foo HTTP/1.1" 403 337
2019-05-24 14:12:32,263 ERROR: Failed to autoscale : namespaces "foo" is forbidden: User "system:serviceaccount:kube-system:kube-downscaler" cannot get resource "namespaces" in API group "" in the namespace "foo"
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/pykube/http.py", line 239, in raise_for_status
    resp.raise_for_status()
  File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://172.20.0.1:443/api/v1/namespaces/foo

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/kube_downscaler/main.py", line 41, in run_loop
    dry_run=dry_run, grace_period=grace_period, downtime_replicas=downtime_replicas)
  File "/kube_downscaler/scaler.py", line 164, in scale
    default_uptime, default_downtime, forced_uptime, dry_run, now, grace_period, downtime_replicas)
  File "/kube_downscaler/scaler.py", line 134, in autoscale_resources
    namespace_obj = pykube.Namespace.objects(api).get_by_name(resource.namespace)
  File "/usr/local/lib/python3.7/site-packages/pykube/query.py", line 80, in get_by_name
    self.api.raise_for_status(r)
  File "/usr/local/lib/python3.7/site-packages/pykube/http.py", line 246, in raise_for_status
    raise HTTPError(resp.status_code, payload["message"])
pykube.exceptions.HTTPError: namespaces "foo" is forbidden: User "system:serviceaccount:kube-system:kube-downscaler" cannot get resource "namespaces" in API group "" in the namespace "foo"
```

This in a newer version than is currently in this PR. (ie, trying tag 0.15, rather than 0.6)

Reply:
See [here](https://github.com/hjacobs/kube-downscaler/blob/78007f0cb14147136ffc129d9a1f88fb5d10f585/helm-chart/templates/clusterrole.yaml#L33).

Reply:
FWIW, going to 0.15 and [adding support for namespaces](https://github.com/bambooengineering/charts/commit/fc3c6e2dbd95527d1935a1ba8322a0cca5c1a728) all seems to work great.